python/cudf_polars/cudf_polars/testing/plugin.py

```diff
@@ -181,6 +181,12 @@ def pytest_configure(config: pytest.Config) -> None:
     "tests/unit/operations/test_group_by.py::test_aggregated_scalar_elementwise_15602": "Unsupported boolean function/dtype combination in groupby-agg",
     "tests/unit/operations/test_group_by.py::test_schemas[data1-expr1-expected_select1-expected_gb1]": "Mismatching dtypes, needs cudf#15852",
     "tests/unit/operations/test_join.py::test_cross_join_slice_pushdown": "Need to implement slice pushdown for cross joins",
+    "tests/unit/operations/test_join.py::test_update": 'LazyFrame.update doesn\'t maintain order of either table (ie. maintain_order="none")',
+    "tests/unit/operations/test_join.py::test_join_preserve_order_left": 'The test includes a join w/ maintain_order="none"',
+    "tests/unit/streaming/test_streaming.py::test_streaming_generic_left_and_inner_join_from_disk": 'The test includes a join w/ maintain_order="none"',
+    "tests/unit/streaming/test_streaming_join.py::test_join_null_matches[False]": 'The test includes a join w/ maintain_order="none"',
+    "tests/unit/streaming/test_streaming_join.py::test_join_null_matches_multiple_keys[False]": 'The test includes a join w/ maintain_order="none"',
+    "tests/unit/test_cse.py::test_cse_rename_cross_join_5405": 'The test includes a join w/ maintain_order="none"',
```

Comment:
Should I open an issue in polars to add `maintain_order="left_right"` or some other value than `none`? That way we wouldn't have to xfail some of these tests. WDYT?
cc. @wence-

Reply:
Oh maybe, yes

Reply:
I tested this PR with Polars 1.19 and these tests now pass

Reply:
I think the problem is related to left joins preserving the order of the left table by default. There are two options I see
1. If I'm right about left joins, we can try handling that case in this PR (ie. how="left" and maintain_order=None => preserve the left table order)
2. Keep the PR as is and do a polars version bump
Thoughts?

Reply:
These tests still fail after upgrading polars to 1.21. I rechecked and I was incorrect about the tests passing 1.19. This makes sense because these I think these tests all have joins w/`maintain_order="none"` (ie. no ordering is guaranteed). 

So, I think we still need to xfail these tests until we possibly update those polars tests.
cc. @wence-